src/pczt.rs

```diff
@@ -0,0 +1,406 @@
+
+        // Run the Creator and Constructor roles.
+        let mut builder = Builder::new(
+            BundleType::DEFAULT,
+            EMPTY_ROOTS[MERKLE_DEPTH_ORCHARD].into(),
+        );
+        builder
+            .add_output(None, recipient, NoteValue::from_raw(5000), None)
+            .unwrap();
+        let balance: i64 = builder.value_balance().unwrap();
+        assert_eq!(balance, -5000);
+        let mut pczt_bundle = builder.build_for_pczt(&mut rng).unwrap().0;
+
+        // Run the IO Finalizer role.
+        let sighash = [0; 32];
+        pczt_bundle.finalize_io(sighash, &mut rng).unwrap();
+
+        // Run the Prover role.
+        pczt_bundle.create_proof(&pk, &mut rng).unwrap();
+
+        // Run the Transaction Extractor role.
+        let bundle = pczt_bundle.extract::<i64>().unwrap().unwrap();
+
+        assert_eq!(bundle.value_balance(), &(-5000));
+        // We can successfully bind the bundle.
+        bundle.bind(sighash, &mut rng).unwrap();
+    }
+
+    #[test]
+    fn shielded_bundle() {
+        let pk = ProvingKey::build();
+        let mut rng = OsRng;
+
+        // Pretend we derived the spending key via ZIP 32.
+        let zip32_derivation = Zip32Derivation::parse([1; 32], vec![]).unwrap();
+        let sk = SpendingKey::random(&mut rng);
+        let ask = SpendAuthorizingKey::from(&sk);
+        let fvk = FullViewingKey::from(&sk);
+        let recipient = fvk.address_at(0u32, Scope::External);
+
+        // Pretend we already received a note.
+        let value = NoteValue::from_raw(5000);
+        let note = {
+            let rho = Rho::from_bytes(&pallas::Base::random(&mut rng).to_repr()).unwrap();
+            loop {
+                if let Some(note) =
+                    Note::from_parts(recipient, value, rho, RandomSeed::random(&mut rng, &rho))
+                        .into_option()
+                {
+                    break note;
+                }
+            }
+        };
+
+        // Use the tree with a single leaf.
+        let (anchor, merkle_path) = {
+            let cmx: ExtractedNoteCommitment = note.commitment().into();
+            let leaf = MerkleHashOrchard::from_cmx(&cmx);
+            let mut tree = BridgeTree::<MerkleHashOrchard, u32, 32>::new(100);
+            tree.append(leaf);
+            let position = tree.mark().unwrap();
+            let root = tree.root(0).unwrap();
+            let auth_path = tree.witness(position, 0).unwrap();
+            let merkle_path = MerklePath::from_parts(
+                u64::from(position).try_into().unwrap(),
+                auth_path[..].try_into().unwrap(),
+            );
+            let anchor = root.into();
+            assert_eq!(anchor, merkle_path.root(cmx));
+            (anchor, merkle_path)
+        };
+
+        // Run the Creator and Constructor roles.
+        let mut builder = Builder::new(BundleType::DEFAULT, anchor);
+        builder.add_spend(fvk, note, merkle_path).unwrap();
```

Comment:
Modify this test to send an output to a different recipient (with change), and then verify that the PCZT contains sufficient information to decrypt and check `enc_ciphertext`.